did/peer.go

```diff
@@ -258,22 +258,22 @@ func (d DIDPeer) buildVerificationMethod(data, did string) (*VerificationMethod,
 // Split the DID string into element.
 // Extract element purpose and decode each key or service.
 // Insert each key or service into the document according to the designated pu
-func (m PeerMethod2) Resolve(did DID) (*DIDDocument, error) {
+func (m PeerMethod2) Resolve(did DID, opts ResolutionOptions) (*DIDDocument, *DIDResolutionMetadata, *DIDDocumentMetadata, error) {
```

Comment:
Could you clarify this one?

Reply:
something like

```
type PeerResolution struct {
  *DIDDocument
  *DIDResolutionMetadata
  *DIDDocumentMetadata
} 
func (m PeerMethod2) Resolve(did DID, opts ResolutionOptions) (*PeerResolution, error) {
```

Reply:
I see what you were going for here. 

We can do it this way, but what's different from PeerResolution vs. a standard DIDResolution? I'd like to avoid having redundant code unless there's a good reason to do so, but if there's a good reason I'm for it.

Reply:
Yea...that could be good. I'll submit a PR with that.